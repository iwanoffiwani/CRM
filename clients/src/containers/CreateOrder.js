import React, { useState } from 'react'
import { connect } from 'react-redux'
import axios from 'axios'
import { fetchUpdateOrderList } from '../redux/actions'
import { makeStyles } from '@material-ui/core/styles'
import Container from '@material-ui/core/Container'
import Typography from '@material-ui/core/Typography'
import TextField from '@material-ui/core/TextField'
import Button from '@material-ui/core/Button'
import Divider from '@material-ui/core/Divider'
import MenuItem from '@material-ui/core/MenuItem'
import Select from '@material-ui/core/Select'
import green from '@material-ui/core/colors/green'

const useStyles = makeStyles(theme => ({
  root: {
    padding: '2rem 0'
  },
  error: {
    margin: 0,
    marginBottom: theme.spacing(1)
  },
  success: {
    margin: 0,
    marginBottom: theme.spacing(1),
    color: green[500]
  },
  formControl: {
    margin: 0
  },
  textField: {
    margin: 0,
    marginBottom: theme.spacing(1)
  },
  title: {
    marginBottom: theme.spacing(4),
    fontWeight: 'bold',
    fontSize: '1.5em'
  }, 
  divider: {
    marginTop: theme.spacing(4),
    marginBottom: theme.spacing(6)
  },
  selectEmpty: {
    marginTop: theme.spacing(2),
  },
  hiddenInput: {
    visibility: 'hidden'
  }
}))

const CreateOrder = props => {
  const classes = useStyles()

  const initialState = {
    response: {
      success: {
        status: null,
        message: null
      },
      error: {
        status: null,
        message: null
      }
    },
    query: {
      status: {
        name: props.statuses[0].name
      },
      name: '',
      fields: [...props.fields]
    }
  }

  const [ state, setState ] = useState(initialState)

  const changeNameHandler = e => {
    return setState({
      ...state,
      query: {
        ...state.query,
        name: e.target.value
      }
    })
  }

  const changeStatusHandler = e => {
    return setState({
      ...state,
      query: {
        ...state.query,
        status: {
          name: e.target.value
        }
      }
    })
  }

  const changeFieldsHandler = e => {
    return setState({
      ...state,
      query: {
        ...state.query,
        fields: state.query.fields.map(field => {
          if (field.name === e.target.name) {
            return {
              ...field,
              value: e.target.value
            }
          } else {
            return field
          }
        })
      }
    })
  }

  const submitHandler = e => {
    e.preventDefault()

    return axios.post('/api/orders', state.query)
      .then(res => setState({
        ...initialState,
        response: {
          ...initialState.response,
          success: {
            status: res.status,
            message: res.data
          }
        }
      }))
      .catch(err => setState({ 
        ...initialState,
        response: {
          ...initialState.response,
          error: {
            status: err.response.status,
            message: `Oops, что то пошло не так.`
          }
        }
      }))
      .then(() => props.update())
  }

  const fields = state.query.fields.map((field, index) => 
    <TextField
      className={classes.textField}
      key={index}
      type='text'
      label={field.name}
      name={field.name}
      value={field.value}
      variant='outlined'
      onChange={changeFieldsHandler}
      fullWidth={true}/>
  )

  const statuses = props.statuses.map((status, index) => 
    <MenuItem 
      id={status._id}
      key={index}
      value={status.name}>
      {status.name}
    </MenuItem>
  ) 

  return (
    <form onSubmit={submitHandler}  className={classes.root}>
      <Container>
        {
          state.response.error.status ? 
          <Typography
            color='error'
            className={classes.error}
            variantMapping={{ h3: 'h3' }}
          >{state.response.error.message}</Typography> : false
        }
        {
          state.response.success.status ? 
          <Typography
            color='error'
            className={classes.success}
            variantMapping={{ h3: 'h3' }}
          >{state.response.success.message}</Typography> : false
        }
        <Typography 
          className={classes.title} 
          variant='h2' 
          noWrap>
          Новая заявка
        </Typography>
        <TextField
          label='Название заявки'
          type='text'
          name='name'
          value={state.query.name}
          margin='normal'
          variant='outlined'
          onChange={changeNameHandler}
          fullWidth={true}
          className={classes.textField}
        />
        <Divider className={classes.divider}/>
        <Typography 
          className={classes.title} 
          variant='h3' 
          noWrap>
          Статус заявки
        </Typography >
        <Select
          id="demo-simple-select-outlined"
          value={state.query.status.name}
          variant='outlined'
          onChange={changeStatusHandler}
          fullWidth={true}>
          {statuses}
        </Select>
        <Divider 
          className={classes.divider}/>
        <Typography 
          className={classes.title} 
          variant='h3' 
          noWrap>
          Произвольные поля
        </Typography >
        {fields}
        <Divider 
          className={classes.divider}/>
        {state.query.name.length < 1 ? 
          <Button 
            variant='contained' 
            disabled>
            Добавить заявку
          </Button> : 
          <Button 
          variant='contained' 
          color='primary' 
          type='submit'>
            Добавить заявку
          </Button>}
      </Container>
    </form>
  )
}

const mapStateToProps = state => {
  return {
    statuses: state.statuses.payload,
    fields: state.fields.payload
  }
}

const mapDispatchToProps = dispatch => {
  return {
    update: () => dispatch(fetchUpdateOrderList())
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(CreateOrder)