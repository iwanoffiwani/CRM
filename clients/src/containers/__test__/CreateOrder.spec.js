import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

import fetchMock from "fetch-mock";
import assert from "assert";

import { CreateOrder } from "../CreateOrder";
import Select from "@material-ui/core/Select";
import TextField from "@material-ui/core/TextField";
import MenuItem from "@material-ui/core/MenuItem";

describe("Tests CreateOrder component", () => {
  const mapStateToProps = {
    statuses: [
      {
        _id: "5e28378cfa502c1edc0e7689",
        name: "Статус-1",
        __v: 0
      },
      {
        _id: "5e283799fa502c1edc0e768b",
        name: "Статус-2",
        __v: 0
      },
      {
        _id: "5e28379ffa502c1edc0e768c",
        name: "Статус-4",
        __v: 0
      },
      {
        _id: "5e2837a5fa502c1edc0e768d",
        name: "Статус-5",
        __v: 0
      }
    ],
    fields: [
      {
        value: "",
        _id: "5dcac8a093dd3b05f8cb0adc",
        name: "Мобильный телефон",
        __v: 0
      },
      {
        value: "",
        _id: "5df9f19e1607be20381256e8",
        name: "Номер паспорта",
        __v: 0
      },
      {
        value: "",
        _id: "5dfa29e98ef6f51a186c395e",
        name: "Домашний тел",
        __v: 0
      },
      {
        value: "",
        _id: "5e00b1bfa63229263405fd21",
        name: "Паспорт",
        __v: 0
      }
    ]
  };

  it("+++ renderer component", () => {
    const wrapper = shallow(<CreateOrder {...mapStateToProps} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("+++ simulation of a change event on the text field(order name)", () => {
    const wrapper = shallow(<CreateOrder {...mapStateToProps} />);

    const textFieldWrapper = wrapper.find(TextField).first();
    textFieldWrapper.simulate("change", {
      target: { value: "Some name for order" }
    });

    expect(
      wrapper
        .find(TextField)
        .first()
        .props().value
    ).toBe("Some name for order");
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("+++ count the number of statuses and fields", () => {
    const wrapper = shallow(<CreateOrder {...mapStateToProps} />);

    const statusesWrapper = wrapper.find(MenuItem);
    expect(statusesWrapper.length).toEqual(mapStateToProps.statuses.length);

    const fieldsWrapper = wrapper.find(TextField);
    expect(fieldsWrapper.length).toEqual(mapStateToProps.fields.length + 1);
  });

  it("+++ simulation of a change event on the select status in select field", () => {
    const wrapper = shallow(<CreateOrder {...mapStateToProps} />);

    const selectWrapper = wrapper.find(Select).first();

    const { _id, name: value } = mapStateToProps.statuses[
      mapStateToProps.statuses.length - 1
    ];

    selectWrapper.simulate("change", {
      _targetInst: {
        stateNode: {
          _id
        }
      },
      target: {
        value
      }
    });

    expect(
      wrapper
        .find(Select)
        .first()
        .props().value
    ).toBe(value);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("simulation of a change event on the text field", () => {
    const wrapper = shallow(<CreateOrder {...mapStateToProps} />);

    const textFieldWrapper = wrapper.find(TextField).last();
    const { name } = mapStateToProps.fields[mapStateToProps.fields.length - 1];

    textFieldWrapper.simulate("change", { target: { value: name, name } });

    expect(
      wrapper
        .find(TextField)
        .last()
        .props().value
    ).toBe(name);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("+++ sending data to create an order (success)", async () => {
    fetchMock.mock("/api/orders", {
      body: {
        message: "Ваша заявка успешно добавлена"
      },
      status: 201
    });

    const res = await fetch("/api/orders", {
      method: "POST",
      body: JSON.stringify({
        fields: mapStateToProps.fields,
        name: "Some order name",
        status: {
          ...mapStateToProps.statuses[0]
        }
      })
    });

    assert(res);
    fetchMock.restore();
  });
});
