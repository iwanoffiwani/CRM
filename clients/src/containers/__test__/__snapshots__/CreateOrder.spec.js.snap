// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests CreateOrder component +++ renderer component 1`] = `
<form
  className="makeStyles-root-1"
  onSubmit={[Function]}
>
  <WithStyles(ForwardRef(Container))>
    <WithStyles(ForwardRef(Typography))
      className="makeStyles-title-6"
      noWrap={true}
      variant="h2"
    >
      Новая заявка
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(TextField))
      className="makeStyles-textField-5"
      fullWidth={true}
      label="Название заявки"
      margin="normal"
      name="name"
      onChange={[Function]}
      type="text"
      value=""
      variant="outlined"
    />
    <WithStyles(ForwardRef(Divider))
      className="makeStyles-divider-7"
    />
    <WithStyles(ForwardRef(Typography))
      className="makeStyles-title-6"
      noWrap={true}
      variant="h3"
    >
      Статус заявки
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(Select))
      fullWidth={true}
      id="demo-simple-select-outlined"
      onChange={[Function]}
      value="Статус-1"
      variant="outlined"
    >
      <WithStyles(ForwardRef(MenuItem))
        id="5e28378cfa502c1edc0e7689"
        key="0"
        value="Статус-1"
      >
        Статус-1
      </WithStyles(ForwardRef(MenuItem))>
      <WithStyles(ForwardRef(MenuItem))
        id="5e283799fa502c1edc0e768b"
        key="1"
        value="Статус-2"
      >
        Статус-2
      </WithStyles(ForwardRef(MenuItem))>
      <WithStyles(ForwardRef(MenuItem))
        id="5e28379ffa502c1edc0e768c"
        key="2"
        value="Статус-4"
      >
        Статус-4
      </WithStyles(ForwardRef(MenuItem))>
      <WithStyles(ForwardRef(MenuItem))
        id="5e2837a5fa502c1edc0e768d"
        key="3"
        value="Статус-5"
      >
        Статус-5
      </WithStyles(ForwardRef(MenuItem))>
    </WithStyles(ForwardRef(Select))>
    <WithStyles(ForwardRef(Divider))
      className="makeStyles-divider-7"
    />
    <WithStyles(ForwardRef(Typography))
      className="makeStyles-title-6"
      noWrap={true}
      variant="h3"
    >
      Произвольные поля
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(TextField))
      className="makeStyles-textField-5"
      fullWidth={true}
      key="0"
      label="Мобильный телефон"
      name="Мобильный телефон"
      onChange={[Function]}
      type="text"
      value=""
      variant="outlined"
    />
    <WithStyles(ForwardRef(TextField))
      className="makeStyles-textField-5"
      fullWidth={true}
      key="1"
      label="Номер паспорта"
      name="Номер паспорта"
      onChange={[Function]}
      type="text"
      value=""
      variant="outlined"
    />
    <WithStyles(ForwardRef(TextField))
      className="makeStyles-textField-5"
      fullWidth={true}
      key="2"
      label="Домашний тел"
      name="Домашний тел"
      onChange={[Function]}
      type="text"
      value=""
      variant="outlined"
    />
    <WithStyles(ForwardRef(TextField))
      className="makeStyles-textField-5"
      fullWidth={true}
      key="3"
      label="Паспорт"
      name="Паспорт"
      onChange={[Function]}
      type="text"
      value=""
      variant="outlined"
    />
    <WithStyles(ForwardRef(Divider))
      className="makeStyles-divider-7"
    />
    <WithStyles(ForwardRef(Button))
      disabled={true}
      variant="contained"
    >
      Добавить заявку
    </WithStyles(ForwardRef(Button))>
  </WithStyles(ForwardRef(Container))>
</form>
`;

exports[`Tests CreateOrder component +++ simulation of a change event on the select status in select field 1`] = `
<form
  className="makeStyles-root-1"
  onSubmit={[Function]}
>
  <WithStyles(ForwardRef(Container))>
    <WithStyles(ForwardRef(Typography))
      className="makeStyles-title-6"
      noWrap={true}
      variant="h2"
    >
      Новая заявка
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(TextField))
      className="makeStyles-textField-5"
      fullWidth={true}
      label="Название заявки"
      margin="normal"
      name="name"
      onChange={[Function]}
      type="text"
      value=""
      variant="outlined"
    />
    <WithStyles(ForwardRef(Divider))
      className="makeStyles-divider-7"
    />
    <WithStyles(ForwardRef(Typography))
      className="makeStyles-title-6"
      noWrap={true}
      variant="h3"
    >
      Статус заявки
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(Select))
      fullWidth={true}
      id="demo-simple-select-outlined"
      onChange={[Function]}
      value="Статус-5"
      variant="outlined"
    >
      <WithStyles(ForwardRef(MenuItem))
        id="5e28378cfa502c1edc0e7689"
        key="0"
        value="Статус-1"
      >
        Статус-1
      </WithStyles(ForwardRef(MenuItem))>
      <WithStyles(ForwardRef(MenuItem))
        id="5e283799fa502c1edc0e768b"
        key="1"
        value="Статус-2"
      >
        Статус-2
      </WithStyles(ForwardRef(MenuItem))>
      <WithStyles(ForwardRef(MenuItem))
        id="5e28379ffa502c1edc0e768c"
        key="2"
        value="Статус-4"
      >
        Статус-4
      </WithStyles(ForwardRef(MenuItem))>
      <WithStyles(ForwardRef(MenuItem))
        id="5e2837a5fa502c1edc0e768d"
        key="3"
        value="Статус-5"
      >
        Статус-5
      </WithStyles(ForwardRef(MenuItem))>
    </WithStyles(ForwardRef(Select))>
    <WithStyles(ForwardRef(Divider))
      className="makeStyles-divider-7"
    />
    <WithStyles(ForwardRef(Typography))
      className="makeStyles-title-6"
      noWrap={true}
      variant="h3"
    >
      Произвольные поля
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(TextField))
      className="makeStyles-textField-5"
      fullWidth={true}
      key="0"
      label="Мобильный телефон"
      name="Мобильный телефон"
      onChange={[Function]}
      type="text"
      value=""
      variant="outlined"
    />
    <WithStyles(ForwardRef(TextField))
      className="makeStyles-textField-5"
      fullWidth={true}
      key="1"
      label="Номер паспорта"
      name="Номер паспорта"
      onChange={[Function]}
      type="text"
      value=""
      variant="outlined"
    />
    <WithStyles(ForwardRef(TextField))
      className="makeStyles-textField-5"
      fullWidth={true}
      key="2"
      label="Домашний тел"
      name="Домашний тел"
      onChange={[Function]}
      type="text"
      value=""
      variant="outlined"
    />
    <WithStyles(ForwardRef(TextField))
      className="makeStyles-textField-5"
      fullWidth={true}
      key="3"
      label="Паспорт"
      name="Паспорт"
      onChange={[Function]}
      type="text"
      value=""
      variant="outlined"
    />
    <WithStyles(ForwardRef(Divider))
      className="makeStyles-divider-7"
    />
    <WithStyles(ForwardRef(Button))
      disabled={true}
      variant="contained"
    >
      Добавить заявку
    </WithStyles(ForwardRef(Button))>
  </WithStyles(ForwardRef(Container))>
</form>
`;

exports[`Tests CreateOrder component +++ simulation of a change event on the text field(order name) 1`] = `
<form
  className="makeStyles-root-1"
  onSubmit={[Function]}
>
  <WithStyles(ForwardRef(Container))>
    <WithStyles(ForwardRef(Typography))
      className="makeStyles-title-6"
      noWrap={true}
      variant="h2"
    >
      Новая заявка
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(TextField))
      className="makeStyles-textField-5"
      fullWidth={true}
      label="Название заявки"
      margin="normal"
      name="name"
      onChange={[Function]}
      type="text"
      value="Some name for order"
      variant="outlined"
    />
    <WithStyles(ForwardRef(Divider))
      className="makeStyles-divider-7"
    />
    <WithStyles(ForwardRef(Typography))
      className="makeStyles-title-6"
      noWrap={true}
      variant="h3"
    >
      Статус заявки
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(Select))
      fullWidth={true}
      id="demo-simple-select-outlined"
      onChange={[Function]}
      value="Статус-1"
      variant="outlined"
    >
      <WithStyles(ForwardRef(MenuItem))
        id="5e28378cfa502c1edc0e7689"
        key="0"
        value="Статус-1"
      >
        Статус-1
      </WithStyles(ForwardRef(MenuItem))>
      <WithStyles(ForwardRef(MenuItem))
        id="5e283799fa502c1edc0e768b"
        key="1"
        value="Статус-2"
      >
        Статус-2
      </WithStyles(ForwardRef(MenuItem))>
      <WithStyles(ForwardRef(MenuItem))
        id="5e28379ffa502c1edc0e768c"
        key="2"
        value="Статус-4"
      >
        Статус-4
      </WithStyles(ForwardRef(MenuItem))>
      <WithStyles(ForwardRef(MenuItem))
        id="5e2837a5fa502c1edc0e768d"
        key="3"
        value="Статус-5"
      >
        Статус-5
      </WithStyles(ForwardRef(MenuItem))>
    </WithStyles(ForwardRef(Select))>
    <WithStyles(ForwardRef(Divider))
      className="makeStyles-divider-7"
    />
    <WithStyles(ForwardRef(Typography))
      className="makeStyles-title-6"
      noWrap={true}
      variant="h3"
    >
      Произвольные поля
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(TextField))
      className="makeStyles-textField-5"
      fullWidth={true}
      key="0"
      label="Мобильный телефон"
      name="Мобильный телефон"
      onChange={[Function]}
      type="text"
      value=""
      variant="outlined"
    />
    <WithStyles(ForwardRef(TextField))
      className="makeStyles-textField-5"
      fullWidth={true}
      key="1"
      label="Номер паспорта"
      name="Номер паспорта"
      onChange={[Function]}
      type="text"
      value=""
      variant="outlined"
    />
    <WithStyles(ForwardRef(TextField))
      className="makeStyles-textField-5"
      fullWidth={true}
      key="2"
      label="Домашний тел"
      name="Домашний тел"
      onChange={[Function]}
      type="text"
      value=""
      variant="outlined"
    />
    <WithStyles(ForwardRef(TextField))
      className="makeStyles-textField-5"
      fullWidth={true}
      key="3"
      label="Паспорт"
      name="Паспорт"
      onChange={[Function]}
      type="text"
      value=""
      variant="outlined"
    />
    <WithStyles(ForwardRef(Divider))
      className="makeStyles-divider-7"
    />
    <WithStyles(ForwardRef(Button))
      color="primary"
      type="submit"
      variant="contained"
    >
      Добавить заявку
    </WithStyles(ForwardRef(Button))>
  </WithStyles(ForwardRef(Container))>
</form>
`;

exports[`Tests CreateOrder component simulation of a change event on the text field 1`] = `
<form
  className="makeStyles-root-1"
  onSubmit={[Function]}
>
  <WithStyles(ForwardRef(Container))>
    <WithStyles(ForwardRef(Typography))
      className="makeStyles-title-6"
      noWrap={true}
      variant="h2"
    >
      Новая заявка
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(TextField))
      className="makeStyles-textField-5"
      fullWidth={true}
      label="Название заявки"
      margin="normal"
      name="name"
      onChange={[Function]}
      type="text"
      value=""
      variant="outlined"
    />
    <WithStyles(ForwardRef(Divider))
      className="makeStyles-divider-7"
    />
    <WithStyles(ForwardRef(Typography))
      className="makeStyles-title-6"
      noWrap={true}
      variant="h3"
    >
      Статус заявки
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(Select))
      fullWidth={true}
      id="demo-simple-select-outlined"
      onChange={[Function]}
      value="Статус-1"
      variant="outlined"
    >
      <WithStyles(ForwardRef(MenuItem))
        id="5e28378cfa502c1edc0e7689"
        key="0"
        value="Статус-1"
      >
        Статус-1
      </WithStyles(ForwardRef(MenuItem))>
      <WithStyles(ForwardRef(MenuItem))
        id="5e283799fa502c1edc0e768b"
        key="1"
        value="Статус-2"
      >
        Статус-2
      </WithStyles(ForwardRef(MenuItem))>
      <WithStyles(ForwardRef(MenuItem))
        id="5e28379ffa502c1edc0e768c"
        key="2"
        value="Статус-4"
      >
        Статус-4
      </WithStyles(ForwardRef(MenuItem))>
      <WithStyles(ForwardRef(MenuItem))
        id="5e2837a5fa502c1edc0e768d"
        key="3"
        value="Статус-5"
      >
        Статус-5
      </WithStyles(ForwardRef(MenuItem))>
    </WithStyles(ForwardRef(Select))>
    <WithStyles(ForwardRef(Divider))
      className="makeStyles-divider-7"
    />
    <WithStyles(ForwardRef(Typography))
      className="makeStyles-title-6"
      noWrap={true}
      variant="h3"
    >
      Произвольные поля
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(TextField))
      className="makeStyles-textField-5"
      fullWidth={true}
      key="0"
      label="Мобильный телефон"
      name="Мобильный телефон"
      onChange={[Function]}
      type="text"
      value=""
      variant="outlined"
    />
    <WithStyles(ForwardRef(TextField))
      className="makeStyles-textField-5"
      fullWidth={true}
      key="1"
      label="Номер паспорта"
      name="Номер паспорта"
      onChange={[Function]}
      type="text"
      value=""
      variant="outlined"
    />
    <WithStyles(ForwardRef(TextField))
      className="makeStyles-textField-5"
      fullWidth={true}
      key="2"
      label="Домашний тел"
      name="Домашний тел"
      onChange={[Function]}
      type="text"
      value=""
      variant="outlined"
    />
    <WithStyles(ForwardRef(TextField))
      className="makeStyles-textField-5"
      fullWidth={true}
      key="3"
      label="Паспорт"
      name="Паспорт"
      onChange={[Function]}
      type="text"
      value="Паспорт"
      variant="outlined"
    />
    <WithStyles(ForwardRef(Divider))
      className="makeStyles-divider-7"
    />
    <WithStyles(ForwardRef(Button))
      disabled={true}
      variant="contained"
    >
      Добавить заявку
    </WithStyles(ForwardRef(Button))>
  </WithStyles(ForwardRef(Container))>
</form>
`;
